base/float.jl

```diff
@@ -349,16 +484,22 @@ end
 <=(x::Float32, y::Union{Int32,UInt32}) = Float64(x)<=Float64(y)
 <=(x::Union{Int32,UInt32}, y::Float32) = Float64(x)<=Float64(y)
 
-abs(x::Float64) = box(Float64,abs_float(unbox(Float64,x)))
+
+abs(x::Float16) = reinterpret(Float16, reinterpret(UInt16,x) & 0x7fff)
 abs(x::Float32) = box(Float32,abs_float(unbox(Float32,x)))
+abs(x::Float64) = box(Float64,abs_float(unbox(Float64,x)))
 
 isnan(x::AbstractFloat) = x != x
+isnan(x::Float16)    = reinterpret(UInt16,x)&0x7fff  > 0x7c00
 isnan(x::Real) = false
 
 isfinite(x::AbstractFloat) = x - x == 0
+isfinite(x::Float16) = reinterpret(UInt16,x)&0x7c00 != 0x7c00
 isfinite(x::Real) = decompose(x)[3] != 0
 isfinite(x::Integer) = true
 
+
+isinf(x::Float16) = reinterpret(UInt16,x)&0x7fff == 0x7c00
```

Comment:
That is correct! With isnan and isfinite defined, this definition is unnecessary.